File: Arrays& Hashing/1.Two sum.java
class Solution{
    public int[] twoSum(int[] nums, int target){
        Map<Integer,Integer> map = new HashMap<>();
        
        for(int j=0; j<nums.length; j++){
            int complement = target-nums[j];
            if(map.containsKey(complement)){
              return new int[]{map.get(complement),j};
            }
            map.put(nums[j],j);
        }
        return new int[]{};
    }
}

File: Arrays& Hashing/1.Two sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for(int j=0; j<nums.size(); j++){
            int complement = target-nums[j];
            if(mp.contains(complement)) return {mp[complement], j};
            mp[nums[j]]=j;
        }
        return {};
    }
};
